Assets/Scripts/UIStateMachine/UIStateMachine.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+
+public static class UIStateMachine
+{
+    public static UIStateMachineBehavior OwningUIStateMachine { get; private set; }
+
+    private static List<UIBaseState> _menuStates = new();
+
+    private static UIBaseState _currentMenu;
+
+    public static List<UIBaseState> MenuStates => _menuStates;
+
+    public static void Subscribe(this UIBaseState uiBaseState)
+    {
+        if(!_menuStates.Contains(uiBaseState)) _menuStates.Add(uiBaseState);
+    }
+
+    public static void SetOwningStateMachine(this UIStateMachineBehavior uiStateMachineBehavior)
+    {
+        OwningUIStateMachine = uiStateMachineBehavior;
+    }

```

Comment:
```suggestion
    public static void SetOwningStateMachine(this UIStateMachineBehavior uiStateMachineBehavior) => OwningUIStateMachine = uiStateMachineBehavior;
```
Misschien is de line te lang dus hoeft niet perse.

Reply:
Expres zo gedaan wegens dat het anders te lang is